### PR TITLE
Post-fix Exprssion calculator added

### PR DESCRIPTION
**Pull request by Rameshwar3297**
#### Description of Change :  Added post-fix expression calculator function #853 

- [X]  Added description of change

- [X] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
Screenshot from 2020-06-15 13-52-53

- [x] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
 Rameshwar3297 Post-fix calculator is added 
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes: 
I have added Post-fix expression calculator,And its working fine, correct output as expected

### CODE REVIEW

File: math/Postfix_Calculator.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
+
```
Comment (kvedala) [inline math/Postfix_Calculator.cpp:1]: This is a redundant file and can be deleted

File: math/postfix_calculator.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
+/*
```
Comment (kvedala) [inline math/postfix_calculator.cpp:1]: Please add comments in [doxygen](https://www.doxygen.nl/manual/docblocks.html) style. You can also refer to #807 and #840

File: math/postfix_calculator.cpp [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+bool is_operand(char ch);
+int priority(char opr);
+int postfix_evaluation(char *post);
+int calculate(int op1, int op2, char opr);
+
+void infix_to_postfix(char *in, char *post) {
+int j=0;
+std::stack<int> s;
+
+for(int i = 0 ; in[i] != '\0' ; i++ ) {
```
Comment (kvedala) [inline math/postfix_calculator.cpp:23]: code is not structured with spaces. Follow Google syntax style. If you use VSCode, the repository already has a syntax style format specifier for it

File: math/postfix_calculator.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+void infix_to_postfix(char *in, char *post);
+bool is_operand(char ch);
+int priority(char opr);
+int postfix_evaluation(char *post);
+int calculate(int op1, int op2, char opr);
+
+void infix_to_postfix(char *in, char *post) {
+int j=0;
+std::stack<int> s;
```
Comment (kvedala) [inline math/postfix_calculator.cpp:21]: avoid global variables and initialize them within functions, pass as arguments

File: math/postfix_calculator.cpp [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+case '+':
+case '-': return 1;
+case '*':
+case '/':
+case '%': return 2;
+}
+return 0;
+}
+
+int main(void) {
```
Comment (kvedala) [inline math/postfix_calculator.cpp:92]: none of the functions are documented.

File: math/Postfix_Calculator.cpp [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+#include<iostream>
+using namespace std;
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:3]: Please don't use namespace using-directives.
Use using-declarations instead.

File: math/Postfix_Calculator.cpp (hunk 1 above)
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:1]: Please add a copyright notice like in #807.

File: math/Postfix_Calculator.cpp [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+//Declaration of the functions
+void infix_to_postfix(char *in, char *post);
+bool is_operand(char ch);
+int priority(char opr);
+int postfix_evaluation(char *post);
+int calculate(int op1, int op2, char opr);
+
+//cur = current
+
+void infix_to_postfix(char *in, char *post)
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:16]: Undocumented function.

File: math/Postfix_Calculator.cpp [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+		if( is_operand(in[i]) )
+		{
+			//append it into the postfix string
+			post[j++] = in[i];
+		}
+		else
+		if( in[i] == '(')//if the current is an opening brace
+		{
+			//push it into the stack
+			s.push(in[i]);
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:33]:
```suggestion
			s.push_back(in[i]);
```
Same in other parts of the code.

File: math/Postfix_Calculator.cpp [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+	{
+		post[j++] = s.top();
+		s.pop();
+	}
+
+	post[j] = '\0';
+}
+
+
+int postfix_evaluation(char *post)
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:75]: Function not documented.

File: math/Postfix_Calculator.cpp [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+	switch(opr)
+	{
+	case '(': return 0;
+
+	case '+':
+	case '-': return 1;
+
+	case '*':
+	case '/':
+	case '%': return 2;
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:140]: Missing `break`s:

```c++
case '-': return 1; break;
```

File: math/Postfix_Calculator.cpp [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	return 0;
+}
+
+
+
+int main(void)
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:147]: Please add a comment structure like:

```c++
/**
 * Main function
 */
```

File: math/Postfix_Calculator.cpp [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+		}
+	}
+
+	//pop the final result from the stack and return it to the calling function
+	result = s.top(); s.pop();
+	return result;
+}
+
+
+bool is_operand(char ch)
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:110]: Function not documented.

File: math/Postfix_Calculator.cpp [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+	case '+': return ( op1 + op2 );
+	case '-': return ( op1 - op2 );
+	case '*': return ( op1 * op2 );
+	case '/': return ( op1 / op2 );
+	case '%': return ( op1 % op2 );
+	}
+return 0;
+}
+
+//we are defining priority for a an operators - programmatically
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:128]: Please use a comment structure like:

```c++
/**
 * Something here.
 * Something else here.
 */
```

File: math/Postfix_Calculator.cpp [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	return 0;
+}
+
+
+
+int main(void)
+{
+	char infix[64];
+	cout<<"Enter Infix Expression :";
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:150]: Use `std::cout`.

File: math/Postfix_Calculator.cpp [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+	cin>>infix;
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:151]:
```suggestion
	std::cin.getline(infix, 64, '\n');
```

File: math/Postfix_Calculator.cpp [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+	char postfix[64] = "";
+
+	cout<<endl;
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:154]:
```suggestion
	std::cout<<std::endl;
```

File: math/Postfix_Calculator.cpp [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+	cout << "Infix Expression : " << infix << endl;
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:155]:
```suggestion
	std::cout << "Infix Expression : " << infix << std::endl;
```

File: math/Postfix_Calculator.cpp [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+	infix_to_postfix(infix, postfix);
+	cout << "Postfix Expression : " << postfix << endl;
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:157]:
```suggestion
	std::cout << "Postfix Expression : " << postfix << std::endl;
```

File: math/Postfix_Calculator.cpp [hunk 19]
```diff
@@ -0,0 +1,170 @@
...
+
+	cout << "Postfix Evaluation Result = " << postfix_evaluation(postfix) << endl;
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:159]:
```suggestion
	std::cout << "Postfix Evaluation Result = " << postfix_evaluation(postfix) << std::endl;
```

File: math/Postfix_Calculator.cpp [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+
+	return 0;
+}
+
+
+
+
+
+
+
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:170]: Please remove all these empty lines (only keep one at the end).

File: math/Postfix_Calculator.cpp [hunk 21]
```diff
@@ -0,0 +1,170 @@
...
+int postfix_evaluation(char *post)
+{
+	int result;
+	stack<int> s;
+
+	for( int i = 0 ; post[i] != '\0' ; i++ )
+	{
+		//if the cur ele is an operand
+		if( is_operand(post[i]))
```
Comment (Panquesito7) [inline math/Postfix_Calculator.cpp:83]: Please move brackets one line above:

```c++
if (var1 == var2) {
    // something here
} else
    // something here
}
```
Same in various other parts of the code.

File: math/postfix_calculator.cpp [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+     char infix[64];
+     std::cout << "Enter Infix Expression :";
+     std::cin >> infix;
```
Comment (Panquesito7) [inline math/postfix_calculator.cpp:95]:
```suggestion
     std::cin.getline(infix, 64, '\n');
```

File: math/postfix_calculator.cpp [hunk 23]
```diff
@@ -0,0 +1,104 @@
...
+     char postfix[64] = "";
+     std::cout << std::endl;
+     std::cout << "Infix Expression : " << infix << std::endl;
+     infix_to_postfix(infix, postfix);
+     std::cout << "Postfix Expression : " << postfix << std::endl;
+     std::cout << "Postfix Evaluation Result = " << postfix_evaluation(postfix) << std::endl;
+return 0;
+}
+
```
Comment (Panquesito7) [inline math/postfix_calculator.cpp:104]: Please remove this empty line.

**Comment by Rameshwar3297** [comment]
i did all needed changes ,still it not working

File: math/postfix_calculator.cpp [hunk 24]
```diff
@@ -1,94 +1,121 @@
...
-case '%': return 2;
-}
-return 0;
+    switch(opr) {
+        case '(': return 0;
+        case '+':
+        case '-': return 1;
+        case '*':
+        case '/':
+        case '%': return 2;
```
Comment (mishraabhinn) [inline math/postfix_calculator.cpp:112]: defaut: is missing

File: math/postfix_calculator.cpp [hunk 25]
```diff
@@ -0,0 +1,134 @@
...
+ * calculate function to perform operation on 2 operands
+ * function returns value calculated
+ */
+int calculate(int op1, int op2, char opr) {
+    switch ( opr ) {
+        case '+': return ( op1 + op2 );
+        case '-': return ( op1 - op2 );
+        case '*': return ( op1 * op2 );
+        case '/': return ( op1 / op2 );
+        case '%': return ( op1 % op2 );
```
Comment (Panquesito7) [inline math/postfix_calculator.cpp:96]: Again, missing `break`s.

**Comment by Rameshwar3297** [comment]
There is no need of break.... returning the value from it so......

**Pull request closed without merging**